algorithms.py

```diff
@@ -145,7 +178,100 @@ def configure(self, data, args):
         params = super(XgbGPUHistAlgorithm, self).configure(data, args)
         params.update({"tree_method": "gpu_hist", "gpu_id": 0})
         return params
+        
+class SkRandomForestAlgorithm(Algorithm):
+    def configure(self, data, args):
+        params = shared_params.copy()
+        if data.learning_task == LearningTask.REGRESSION:
+            params["objective"] = "regression"
+        elif data.learning_task == LearningTask.CLASSIFICATION:
+            params["objective"] = "binary"
+            params["scale_pos_weight"] = len(data.y_train) / np.count_nonzero(data.y_train)
+        elif data.learning_task == LearningTask.MULTICLASS_CLASSIFICATION:
+            params["objective"] = "multiclass"
+            params["num_class"] = np.max(data.y_test) + 1
+        params.update(args.extra)
+        return params
+
+    def fit(self, data, args):
+        params = self.configure(data, args)
+        if params["objective"] == "regression":
+            with Timer() as t:
+                self.model = skrf_r(max_depth=params["max_depth"]).fit(data.X_train, data.y_train)
+            return t.interval
+        else:
+            with Timer() as t:
+                self.model = skrf(max_depth=params["max_depth"]).fit(data.X_train, data.y_train)
+            return t.interval
+
+    def test(self, data):
+        return self.model.predict(data.X_test)
+    
+    def __exit__(self, exc_type, exc_value, traceback):
+        del self.model
+
+class SkGradientAlgorithm(Algorithm):
+    def configure(self, data, args):
+        params = shared_params.copy()
+        if data.learning_task == LearningTask.REGRESSION:
+            params["objective"] = "regression"
+        elif data.learning_task == LearningTask.CLASSIFICATION:
+            params["objective"] = "binary"
+            params["scale_pos_weight"] = len(data.y_train) / np.count_nonzero(data.y_train)
+        elif data.learning_task == LearningTask.MULTICLASS_CLASSIFICATION:
+            params["objective"] = "multiclass"
+            params["num_class"] = np.max(data.y_test) + 1
```

Comment:
I don't think scikit-learn's GBM estimator makes distinction between binary classification and multi-class classification. So we should treat all classification tasks in a single if case.